### PR TITLE
Use c8 for code coverage

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added c8 to track code coverage.

## Details
I've been wanting to enable code coverage for a long time.  The intention is to use this for local inspection but also load the lcov reports from CI runs to a code coverage service, I'm thinking codecov to start with tags for different testing types.

To test:
 * `npm ci`
 * `npm test`
 * `open covereage/esm-unit/index.html`
 * `open coverage/unit/index.html`
 * `open coverage/integration/index.html`
 * `npm run versioned:major`
 * `open coverage/versioned/index.html`

Each report will be for the types of test.  This will make it very easy going forward to use codecov to load the lcov reports with relevant tags for given node versions and/or types of testing.  I'm not opposed to keeping on .c8rc.json file but if you run `npm test` the coverage would get stomped on for every type of test which is why I had a separate config.

**Note**: I added include/exclude lists for .mjs files.

### CODE REVIEW

**Review by mrickard — APPROVED** [review]

File: .c8rc-esm-unit.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (jmartin4563) [inline .c8rc-esm-unit.json:1]: Would it be better to put these in a folder so they're not polluting the top-level structure?

File: .c8rc-versioned.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (jmartin4563) [inline .c8rc-versioned.json:1]: Is coverage on these tests actually useful? I get the unit/integration tests, but I feel like there will be very limited coverage on these

**Review by jmartin4563 — COMMENTED** [review]
Couple of comments

File: .c8rc-esm-unit.json (hunk 1 above)
Comment (bizob2828) [reply .c8rc-esm-unit.json:1]: I could. They are dot files and typically hidden and we don't publish these with the agent

File: .c8rc-versioned.json (hunk 2 above)
Comment (bizob2828) [reply .c8rc-versioned.json:1]: versioned tests are the only way we get instrumentation coverage.  these are essentially integration tests for all 3rd party libraries.  When we use a code coverage tool all the coverages will be merged together so it'll give a complete picture.  if you run unit/integration tests you'll see the gaping hole in instrumentation.

File: .c8rc-esm-unit.json (hunk 1 above)
Comment (bizob2828) [reply .c8rc-esm-unit.json:1]: my other option could be to just keep 1 config for the common stuff and pass in the reports dir in package.json when calling c8

Comment (jmartin4563) [reply .c8rc-esm-unit.json:1]: They do show up on github though, which is basically this agent's "homepage". I don't feel so strongly enough to really push it further than this though, so I'll leave it to you :)

Comment (jmartin4563) [reply .c8rc-esm-unit.json:1]: > my other option could be to just keep 1 config for the common stuff and pass in the reports dir in package.json when calling c8

I like this idea!

File: .c8rc-versioned.json (hunk 2 above)
Comment (jmartin4563) [reply .c8rc-versioned.json:1]: I wasn't thinking about the mushing together of coverage, this makes more sense now, thanks!

**Review by jmartin4563 — APPROVED** [review]
Hooray for code coverage! 🙌🏻 🎉

**Pull request merged**